spring-beans/src/main/java/org/springframework/beans/factory/support/AbstractBeanFactory.java

```diff
@@ -830,8 +826,10 @@ public String resolveEmbeddedValue(@Nullable String value) {
 			return null;
 		}
 		String result = value;
-		for (StringValueResolver resolver : this.embeddedValueResolvers) {
-			result = resolver.resolveStringValue(result);
+		Iterator<DefaultIgnorableStringValueResolver> resolverIterator = this.embeddedValueResolvers.iterator();
+		while (resolverIterator.hasNext()) {
+			DefaultIgnorableStringValueResolver resolver = resolverIterator.next();
+			result = (resolverIterator.hasNext() ? resolver.resolveStringValueIgnoringDefault(result) : resolver.resolveStringValue(result));
```

Comment:
The gist of this PR. Only the last DefaultIgnorableStringValueResolver will enable 'default value' processing.